src/psyclone/psyGen.py

```diff
@@ -3508,6 +3508,18 @@ def type(self):
         cases. API's with this concept can override this method '''
         return "field"
 
+    @property
+    @abc.abstractmethod
+    def intrinsic_type(self):
```

Comment:
The addition of this property highlights that the name of the immediately preceding `type` method is not helpful :-) Again, I think something like `object_type` or perhaps we should sub-class `Argument` as `ScalarArgument`, `FieldArgument`, `OperatorArgument` etc. etc. I suspect that's the right way to go as we're asking the question "what type is this Argument object?" and the answer is naturally e.g. "a FieldArgument". That's clearly not something for this PR though so would you mind creating an Issue where we can discuss sub-classing `Argument`? Thanks :-)

Reply:
How about the above suggestion of `argument_type` in this PR and then subclassing in the later ones?

Reply:
Renamed `type` to `argument_type` as discussed above. Created issue #859 to discuss sub-classing `Argument`.